ocs_ci/ocs/resources/storage_cluster.py

```diff
@@ -623,6 +628,49 @@ def osd_encryption_verification():
             )
             raise ValueError("OSD is not encrypted")
 
+    if ocs_version > version.VERSION_4_6:
+        log.info("Verify luks header label for encrypted devices")
+        worker_nodes = get_worker_nodes()
+        logging.info("NODES: {}".format(worker_nodes))
+        failures = 0
+        failure_message = ""
+        node_obj = OCP(kind="node")
+
+        for n in worker_nodes:
+            luks_devices = node_obj.exec_oc_debug_cmd(
+                node=n,
+                cmd_list=[
+                    "lsblk -o NAME,TYPE,FSTYPE | grep -E 'disk.*crypto_LUKS' | awk '{print $1}'"
+                ],
+            ).split("\n")
+
+            for luks_device_name in luks_devices:
+                luks_device_name = luks_device_name.strip()
+                if luks_device_name == "":
+                    continue
+                logging.info(
+                    "Checking luks header label on Luks device {} for node {}".format(
+                        luks_device_name, n
+                    )
+                )
+
+                logging.info("Waiting for 5 seconds")
+                time.sleep(5)
+
+                cmd = "cryptsetup luksDump /dev/" + str(luks_device_name)
+                cmd_out = node_obj.exec_oc_debug_cmd(node=n, cmd_list=[cmd])
+
+                if "(no label)" in str(cmd_out) or "(no subsystem)" in str(cmd_out):
+                    failures += 1
+                    failure_message += "\nNo label found on Luks header information for node {}\n".format(
+                        n
+                    )
+
+        if failures != 0:
+            logging.error(failure_message)
+            raise ValueError("Luks header label is not found")
+        logging.info("Luks header info found for all the encrypted osds")
+
```

Comment:
As part of the BZ we should also try deleting the OSD deployment and see if the osd comes up successfully, do you plan to cover that aswell?

Reply:
yeah, I only automated the steps of bz verification mentioned by rachael. I understand we need to cover osd deployment deletion as part of bz but i'm not sure if its mandatory to check that.. if so then it will become a test case right?

Reply:
I think we can cover this scenario in a separate PR